content/5_Plat/Binary_Jump.mdx

```diff
@@ -129,6 +129,41 @@ int main() {
 
 </CPPSection>
 
+<PySection>
+
+```py
+import math
+
+
+n, q = map(int, input().split())
+
+
+def jmp(x: int, k: int) -> int:  # Get the k'th ancestor of x
+	for i in range(LOGMAX):
+		if k & 1 << i:
+			x = up[i][x]
+		if x == -2:
+			break
+	return x
+
+
+LOGMAX = math.ceil(math.log2(n)) + 1
+up = [[-2 for j in range(n)] for i in range(LOGMAX)]
```

Comment:
-2?

Reply:
also if the +1 does what i think it does, put it here like so:
```suggestion
up = [[-2 for j in range(n)] for i in range(LOGMAX + 1)]
```

Reply:
I put -2 because when there is no answer the output is -1. The +1 is already there.